p2p/p2p.go

```diff
@@ -74,17 +79,32 @@ func New(addr, userAgent, peers, privKeyStr string, feederNode bool, bc *blockch
 	// Todo: try to understand what will happen if user passes a multiaddr with p2p public and a private key which doesn't match.
 	// For example, a user passes the following multiaddr: --p2p-addr=/ip4/0.0.0.0/tcp/7778/p2p/(SomePublicKey) and also passes a
 	// --p2p-private-key="SomePrivateKey". However, the private public key pair don't match, in this case what will happen?
-	return NewWithHost(p2pHost, peers, feederNode, bc, snNetwork, log)
+	return NewWithHost(p2pHost, peers, feederNode, bc, snNetwork, log, database)
 }
 
 func NewWithHost(p2phost host.Host, peers string, feederNode bool, bc *blockchain.Blockchain, snNetwork *utils.Network,
-	log utils.SimpleLogger,
+	log utils.SimpleLogger, database db.DB,
 ) (*Service, error) {
-	peersAddrInfoS := []peer.AddrInfo{}
+	var (
+		peersAddrInfoS []peer.AddrInfo
+		storePeers     []peer.AddrInfo
+		err            error
+	)
+
+	if database != nil {
```

Comment:
you can assume it's always present, for interfaces we don't usually pas nil (only in tests and only when we're sure that they won't be called which

Reply:
If you used it in tests then you can just pass `testDB := pebble.NewMemTest(t)`